packages/blocto/src/adapter.ts

```diff
@@ -0,0 +1,140 @@
+
+    constructor(config: BloctoWalletAdapterConfig) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+        this._network = config.network;
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return true;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.connected;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+            const sdk = new BloctoSDK({ solana: { net: this._network } });
+            const wallet = sdk.solana;
+            if (!wallet) throw new WalletNotFoundError();
+
+            if (!wallet.connected) {
+                try {
+                    await wallet.connect();
+                } catch (error) {
+                    throw new WalletConnectionError(error?.message, error);
+                }
+            }
+
+            if (!wallet.accounts[0]) {
+                throw new WalletAccountError('Account not found');
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.accounts[0]);
+            } catch (error) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect');
+        } catch (error) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        if (this._wallet?.connected) {
+            this._publicKey = null;
+
+            try {
+                this._wallet.disconnect();
+            } catch (error) {
+                this.emit('error', new WalletDisconnectionError(error.message, error));
+            } finally {
+                this._wallet = null;
+            }
+
+            this.emit('disconnect');
+        }
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options: SendTransactionOptions = {}
+    ): Promise<TransactionSignature> {
+        try {
+            if (!this._wallet?.connected) {
+                throw new WalletNotConnectedError();
+            }
+
+            // @TODO: remove this when blocto-sdk implements connection injection
+            if (this._wallet.rpc !== (connection as any)?._rpcEndpoint || connection.commitment) {
+                console.warn("@solana/wallet-adapter-blocto: Blocto SDK has not yet supported commitment nor switching network dynamically, your settings on connection object will be ignored.")
+            }
```

Comment:
This is similar to throwing an error because the adapter can't handle the connection as expected, which the dapp is expected to handle at runtime. This is a compromise I don't think is acceptable. I'm also not comfortable with the compromise of printing a console warning, which is worse because the dapp can't see or do anything about it.

The purpose of the wallet-adapter library is to allow wallets to be used interchangeably by dapps and most importantly their users. Not using the connection object provided to send the transaction is a violation of the principle of least surprise. I will wait until this change can be fully implemented to accept this adapter.